cunumeric/module.py

```diff
@@ -1462,7 +1462,7 @@ def check_shape_dtype(
         )
     if ndim > 1 and _builtin_any(
         shape[1:axis] != inp.shape[1:axis]
-        and shape[axis + 1 :] != inp.shape[axis + 1 :]
+        or shape[axis + 1 :] != inp.shape[axis + 1 :]
```

Comment:
Based on "All arguments must have the same dimension size in all dimensions except the target axis", I think it should be  **or** instead of **and** here. @bryevdv Please help to confirm.
This change would cause failures in test_append.py and test_block.py. I'm looking into it. Maybe there is bug in original code.

Reply:
In addition to replace ```and``` by ```or```, a lot of potential bugs are found by this test, which are all marked as xfail. I would create a new PR to enhance function ```check_shape_dtype``` in cunumeric/module.py to fix these bugs. In this PR, we only create/enhance tests. @bryevdv Please help to review these tests. Thanks.